OpenRA.Mods.D2k/Traits/World/D2kResourceRenderer.cs

```diff
@@ -140,7 +140,8 @@ protected override void UpdateRenderedSprite(CPos cell, RendererCellContents con
 			for (var i = 0; i < directions.Length; i++)
 			{
 				var neighbour = cell + directions[i];
-				UpdateRenderedSpriteInner(neighbour, RenderContents[neighbour]);
+				if (World.Map.Contains(neighbour))
```

Comment:
This should be
```suggestion
				if (RenderContents.Contains(neighbour))
```
see https://github.com/abcdefg30/OpenRA/commit/374ece652fd37fd258bf5632ac59b222de754352.

Otherwise resources outside of the playable area are not rendered correctly.